include/fmt/compile.h

```diff
@@ -44,7 +60,7 @@ struct is_compiled_string : std::is_base_of<compiled_string, S> {};
     std::string s = fmt::format(FMT_COMPILE("{}"), 42);
   \endrst
  */
-#define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
+#define FMT_COMPILE(s) FMT_COMPILE_IMPL(s)
```

Comment:
Could it use the same macro as it used to use before to avoid another macro expansion? Probably not because of docs, but maybe there is a solution for that...

Reply:
We'll have an issue with documentation then. I mean if I use
```
#ifdef __cpp_if_constexpr
#  define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)
#else
#  define FMT_COMPILE(s) <error verion>
#endif
```
But if there is another macro like `FMT_CONDITIONAL( __cpp_if_constexpr, FMT_STRING_IMPL(s, fmt::detail::compiled_string), <error verion> )`
Then it might work. But FMT_CONDITIONAL will add inderection level too.

Also, the following version might work but is kind of ugly 
```
#define FMT_COMPILE(s) FMT_STRING_IMPL(s, fmt::detail::compiled_string)

#ifdef __cpp_if_constexpr
#undef FMT_COMPILE
#  define FMT_COMPILE(s) <error verion>
#endif
```

I don't see a problem with extra macro expansion anyway.